fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/BaseTableLoader.java

```diff
@@ -159,6 +167,138 @@ public void loadLogicalTableWithGranularities(
+            throw new RuntimeException(message);
+        }
+
+        // If dependent table not in physical table dictionary, try to recursively build it and put it in the dictionary
+        currentTableDefinition.getDependentTableNames()
+                .forEach(tableName -> physicalTableDictionary.computeIfAbsent(
+                        tableName.asName(),
+                        name -> buildPhysicalTablesWithDependency(
+                                name,
+                                availableTableDefinitions,
+                                dictionaries
+                        )
+                ));
+
+        // Build the current physical table using physical table dictionary to resolve dependency
+        Optional<PhysicalTable> optionalPhysicalTable = currentTableDefinition.build(
+                dictionaries,
+                getDataSourceMetadataService()
+        );
+
+        // If the table definition correctly builds a physical table
+        if (optionalPhysicalTable.isPresent()) {
+            return optionalPhysicalTable.get();
+        } else {
+            String message = String.format(
+                    "Unable to build physical table: %s, might be table definition error", currentTableName);
+
+            LOG.error(message);
+            throw new RuntimeException(message);
+        }
+    }
+
+    /**
+     * Getter for the data source metadata service use for creating physical tables.
+     *
+     * @return the data source metadata service associated with the table loader
+     */
+    protected DataSourceMetadataService getDataSourceMetadataService() {
+        return metadataService;
+    }
+
+    /**
+     * Build a map from physical table name to its table definition.
+     *
+     * @param physicalTableDefinitions  Definitions to build the map from
+     *
+     * @return the map of physical table name to its table definition
+     */
+    private Map<String, PhysicalTableDefinition> buildPhysicalTableDefinitionDictionary(
+            Set<PhysicalTableDefinition> physicalTableDefinitions
+    ) {
+        return physicalTableDefinitions.stream()
+                .collect(Collectors.toMap(definition -> definition.getName().asName(), Function.identity()));
+    }
+
+    /**
+     * Builds a table group.
+     * <p>
+     * Builds and loads the physical tables for the physical table definitions as well.
+     *
+     * @param apiMetrics  The set of metric names surfaced to the api
+     * @param druidMetrics  Names of druid datasource metric columns
+     * @param tableDefinitions  A list of config objects for physical tables
+     * @param dictionaries  The container for all the data dictionaries
+     *
+     * @return A table group binding all the tables for a logical table view together.
+     *
+     * @deprecated does not load table with external dependency, use the other buildDimensionSpanningTableGroup instead
+     */
+    public TableGroup buildDimensionSpanningTableGroup(
```

Comment:
So, rather than duplicating the code, can we just have the deprecated one call the non-deprecated one?

Also, if this is to be deprecated, it should have the `@Deprecated` annotation on the method so that the compiler will flag it.